main.tf

```
@@ -0,0 +1,101 @@
+terraform {
+    required_providers {
+        azurerm = {
+          source = "hashicorp/azurerm"
+          version = ">= 3.8"
+        }
+    }
+    backend "azurerm" {
+        resource_group_name  = "Cohort22_JonLon_ProjectExercise"
+        storage_account_name = "tfstate1147207684"
+        container_name       = "tfstate"
+        key                  = "$env:ARM_ACCESS_KEY"
+    }
+}
+
+provider "azurerm" {
+    features {}
+# ARM_ environmets are use from Pipeline to login automatically
+}
+
+data "azurerm_resource_group" "main" {
+    name = "Cohort22_JonLon_ProjectExercise"
+}
+
+resource "azurerm_service_plan" "main" {
+  name = "${var.PREFIX}-terraformed-asp"
+  location = data.azurerm_resource_group.main.location
+  resource_group_name = data.azurerm_resource_group.main.name
+  os_type = "Linux"
+  sku_name = "B1"
+}
+
+resource "azurerm_linux_web_app" "main" {
+  name = "${var.PREFIX}-${var.APP_NAME}"
+  location = data.azurerm_resource_group.main.location
+  resource_group_name = data.azurerm_resource_group.main.name
+  service_plan_id = azurerm_service_plan.main.id
+  
+  site_config {
+    application_stack {
+      docker_image = "wishy78/todo-app"
+      docker_image_tag = "latest"
+    }
+  }
+  app_settings = {
+    "FLASK_APP" = "${var.FLASK_APP}"
```

Minor but you might see some warnings for this - you can just pass through the variables directly without interpolation if there's no hardcoded part to the string:
```suggestion
    "FLASK_APP" = var.FLASK_APP
```

---

updated